orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/ConfigStageInjectionTransform.java

```diff
@@ -74,6 +77,24 @@ private void replaceStages(PipelineTemplate pipelineTemplate) {
   }
 
   private void injectStages(PipelineTemplate pipelineTemplate) {
-    // TODO rz - injection isn't going to be a first-pass feature
+    List<StageDefinition> configStages = templateConfiguration.getStages()
+      .stream()
+      .filter(s -> s.getInject() != null)
+      .collect(Collectors.toList());
+    List<StageDefinition> templateStages = pipelineTemplate.getStages()
+      .stream()
+      .filter(s -> s.getInject() != null)
+      .collect(Collectors.toList());
+
+    if (!configStages.isEmpty() || !templateStages.isEmpty()) {
+      throw new UnsupportedOperationException("stage injection is not yet supported");
+    }
+  }
+
+  private void assignRefIds(PipelineTemplate pipelineTemplate) {
+    AtomicInteger refId = new AtomicInteger(1);
+    pipelineTemplate.getStages().forEach(s -> {
+      s.setRefId(refId.getAndIncrement());
```

Comment:
refId is actually a string in the JSON and just happens to be done as an incrementing integer in the UI. If there is already some other unique identifier that we have for each stage could probably just use that (and the resulting JSON maybe has some more useful meaning to it when looking at it). If we don't have an already unique identifier at this point I'm totally okay with just using ints